### PR TITLE
Refactor CategoryEditor in Projects page

### PR DESCRIPTION
**Pull request by frozenhelium**
* Rename CategoryEditor to WordCategory
* Make WordCategory tab consistent with Framework tab
* Fix some bugs in Framework tab
* Improve memoization for selecting active framewok and wordCategory

### CODE REVIEW

File: src/views/Project/Details/Frameworks/FrameworkDetail/index.js [hunk 1]
```diff
@@ -88,24 +88,24 @@ export default class Details extends React.PureComponent {
...
-            framework,
-        } = this.props;
+            framework: {
+                id: frameworkId,
+                isAdmin: isFrameworkAdmin,
+            },
+        } = this.state;
 
-        if (!framework.isAdmin) {
+        if (isFrameworkAdmin) {
```
Comment (tnagorra) [inline src/views/Project/Details/Frameworks/FrameworkDetail/index.js]: The behavior has been inverted!

File: src/views/Project/Details/WordCategories/WordCategoryDetail/CloneWordCategoryButton/index.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            )
+            .start();
+    }
+
+    handleClick = () => {
+        this.setState({ showCloneWordCategoryModal: true });
+    }
+
+    handleModalCancelButtonClick = () => {
+        this.setState({ showCloneWordCategoryModal: false });
```
Comment (tnagorra) [inline src/views/Project/Details/WordCategories/WordCategoryDetail/CloneWordCategoryButton/index.js]: Data should be cleared on unmount

File: src/views/Project/Details/WordCategories/WordCategoryDetail/index.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+@connect(mapStateToProps, mapDispatchToProps)
+export default class WordCategoryDetail extends React.PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    constructor(props) {
+        super(props);
+
+        this.state = {};
```
Comment (tnagorra) [inline src/views/Project/Details/WordCategories/WordCategoryDetail/index.js:63]: should set initial pending state

File: src/views/Project/Details/WordCategories/WordCategoryDetail/CloneWordCategoryButton/index.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+                    disabled={disabled}
+                    onClick={this.handleClick}
+                >
+                    { _ts('project.wordCategory', 'cloneButtonTitle') }
+                </AccentButton>
+                { showCloneWordCategoryModal && (
+                    <Modal className={styles.cloneWordCategoryModal}>
+                        <Faram
+                            onChange={this.handleFaramChange}
+                            onValidationFailure={this.handleValidationFailure}
```
Comment (AdityaKhatri) [inline src/views/Project/Details/WordCategories/WordCategoryDetail/CloneWordCategoryButton/index.js:139]: No handleValidationFailure function defined.

File: src/views/Project/Details/WordCategories/WordCategoryDetail/CloneWordCategoryButton/requests/WordCategoryCloneRequest.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    handlePreLoad = () => {
+        this.parent.setState({ pendingWordCategoryClone: true });
+    }
+
+    handlePostLoad = () => {
+        this.parent.setState({ pendingWordCategoryClone: false });
+    }
+
+    handleSuccess = (response) => {
+        this.parent.setState({ showCloneWordCategoryModal: false });
```
Comment (thenav56) [inline src/views/Project/Details/WordCategories/WordCategoryDetail/CloneWordCategoryButton/requests/WordCategoryCloneRequest.js]: Data should be cleared on unmount here too.

**Comment by thenav56** [comment]
Tab title not changed.
image

File: src/views/Project/Details/WordCategories/WordCategoryDetail/index.js (hunk 3 above)
Comment (frozenhelium) [reply src/views/Project/Details/WordCategories/WordCategoryDetail/index.js:63]: The request is not being used (still to be implemented on server side)

**Comment by frozenhelium** [comment]
> Tab title not changed.
> image

Will be implemented later

File: src/views/Project/Details/WordCategories/WordCategoryDetail/CloneWordCategoryButton/requests/WordCategoryCloneRequest.js (hunk 5 above)
Comment (frozenhelium) [reply src/views/Project/Details/WordCategories/WordCategoryDetail/CloneWordCategoryButton/requests/WordCategoryCloneRequest.js]: done

**Pull request merged**